server/src/metrics/actix_web_metrics.rs

```diff
@@ -363,32 +363,31 @@ where
 
         let request_metrics = self.metrics.clone();
         Box::pin(self.service.call(req).map(move |res| {
-            request_metrics
-                .http_server_active_requests
-                .add(-1, &attributes);
-
-            // Ignore actix errors for metrics
-            if let Ok(res) = res {
-                attributes.push(HTTP_RESPONSE_STATUS_CODE.i64(res.status().as_u16() as i64));
-                let response_size = res
-                    .response()
-                    .headers()
-                    .get(CONTENT_LENGTH)
-                    .and_then(|len| len.to_str().ok().and_then(|s| s.parse().ok()))
-                    .unwrap_or(0);
-                request_metrics
-                    .http_server_response_size
-                    .record(response_size, &attributes);
-
-                request_metrics.http_server_duration.record(
-                    timer.elapsed().map(|t| t.as_secs_f64()).unwrap_or_default(),
-                    &attributes,
-                );
-                Ok(res)
-            } else {
-                res
-            }
-        }))
+          let status_code = match &res {
+              Ok(res) => res.status().as_u16() as i64,
+              Err(e) => {
+                e.as_response_error().status_code().as_u16() as i64
+              },
+          };
+      
+          attributes.push(HTTP_RESPONSE_STATUS_CODE.i64(status_code));
+          request_metrics.http_server_active_requests.add(-1, &attributes);
+      
+          let response_size = match &res {
```

Comment:
Personal taste but I think match statements where the error handling is just return a Copy type are better expressed as map/unwrap_or.

You can do this:


            let response_size = res
                .as_ref()
                .map(|res| {
                    res.response()
                        .headers()
                        .get(CONTENT_LENGTH)
                        .and_then(|len| len.to_str().ok().and_then(|s| s.parse().ok()))
                        .unwrap_or(0u64)
                })
                .unwrap_or(0);

Reply:
Addressed in https://github.com/Unleash/unleash-edge/pull/414/commits/ff0a6184b1dee4dd46be7eb577a66833025c433c